liquidator/index.js

```diff
@@ -189,69 +195,73 @@ async function run(
+      // Read price feed configuration from an environment variable. This can be a crypto watch, medianizer or uniswap
+      // price feed Config defines the exchanges to use. If not provided then the bot will try and infer a price feed
+      // from the EMP_ADDRESS. EG with medianizer: {"type":"medianizer","pair":"ethbtc",
+      // "lookback":7200, "minTimeBetweenUpdates":60,"medianizedFeeds":[{"type":"cryptowatch","exchange":"coinbase-pro"},
+      // {"type":"cryptowatch","exchange":"binance"}]}
+      priceFeedConfig: process.env.PRICE_FEED_CONFIG ? JSON.parse(process.env.PRICE_FEED_CONFIG) : null,
+      // If there is a liquidator config, add it. Else, set to null. This config contains crThreshold,liquidationDeadline,
+      // liquidationMinPrice, txnGasLimit & logOverrides. Example config:
+      // {"crThreshold":0.02,  -> Liquidate if a positions collateral falls more than this % below the min CR requirement
+      //   "liquidationDeadline":300, -> Aborts if the transaction is mined this amount of time after the last update
+      //   "liquidationMinPrice":0, -> Aborts if the amount of collateral in the position per token is below this ratio
+      //   "txnGasLimit":9000000 -> Gas limit to set for sending on-chain transactions.
+      //   "logOverrides":{"positionLiquidated":"warn"}} -> override specific events log levels.
+      liquidatorConfig: process.env.LIQUIDATOR_CONFIG ? JSON.parse(process.env.LIQUIDATOR_CONFIG) : null,
+      // If there is a LIQUIDATOR_OVERRIDE_PRICE environment variable then the liquidator will disregard the price from the
+      // price feed and preform liquidations at this override price. Use with caution as wrong input could cause invalid liquidations.
+      liquidatorOverridePrice: process.env.LIQUIDATOR_OVERRIDE_PRICE
+    };
 
-    await run(
-      Logger,
-      process.env.EMP_ADDRESS,
-      pollingDelay,
-      errorRetries,
-      errorRetriesTimeout,
-      priceFeedConfig,
-      liquidatorConfig,
-      liquidatorOverridePrice
-    );
+    // Check if the bot is being run as a node process or as a truffle process.
+    if (typeof web3 == "undefined") {
+      // Create a web3 instance. This has built in re-try on error and loads in a provided mnemonic or private key.
+      const { web3 } = require("@umaprotocol/financial-templates-lib/clients/Web3WebsocketClient");
+      await run(Logger, web3, ...Object.values(executionParameters));
+      // Else, if the web3 instance is not undefined, then the script is being run from Truffle. Use present web3 instance.
+    } else {
+      await run(Logger, web3, ...Object.values(executionParameters));
+    }
   } catch (error) {
     Logger.error({
       at: "Liquidator#index",
       message: "Liquidator configuration error🚨",
       error: typeof error === "string" ? new Error(error) : error
     });
     await waitForLogger(Logger);
-    callback(error);
-    return;
+    process.exit(1);
   }
-  callback();
```

Comment:
Does this work for truffle? I thought the lack of a callback in the success case would cause truffle to hang? Do you need a `process.exit(0)` here to match the above solution?

Reply:
It does work with truffle. The return statement would get it back to a point where this yields with a `process.exit(0)` further down.

However, I've re-worked this whole execution pattern in accordance with your other comment below.